HidPkg/UefiHidDxe/src/main.rs

```diff
@@ -0,0 +1,58 @@
+//! HID input driver for UEFI
+//!
+//! This crate provides input handlers for HID 1.1 compliant keyboards and pointers.
+//!
+//! ## License
+//!
+//! Copyright (c) Microsoft Corporation. All rights reserved.
+//!
+
+#![no_std]
+#![no_main]
+#![allow(non_snake_case)]
+
+extern crate alloc;
+
+use core::panic::PanicInfo;
+
+use driver_binding::initialize_driver_binding;
+use r_efi::{efi, system};
+
+use rust_advanced_logger_dxe::{debugln, init_debug, DEBUG_ERROR};
+use rust_boot_services_allocator_dxe::GLOBAL_ALLOCATOR;
+
+mod driver_binding;
+mod hid;
+mod keyboard;
+mod pointer;
+
+static mut BOOT_SERVICES: *mut system::BootServices = core::ptr::null_mut();
+
+#[no_mangle]
+pub extern "efiapi" fn efi_main(image_handle: efi::Handle, system_table: *const system::SystemTable) -> efi::Status {
+  // Safety: This block is unsafe because it assumes that system_table and (*system_table).boot_services are correct,
+  // and because it mutates/accesses the global BOOT_SERVICES static.
+  unsafe {
+    BOOT_SERVICES = (*system_table).boot_services;
+    GLOBAL_ALLOCATOR.init(BOOT_SERVICES);
+    init_debug(BOOT_SERVICES);
+  }
+
+  let status = initialize_driver_binding(image_handle);
+
+  if status.is_err() {
+    debugln!(DEBUG_ERROR, "[UefiHidMain]: failed to initialize driver binding.\n");
+  }
+
+  efi::Status::SUCCESS
+}
+
+//Workaround for https://github.com/rust-lang/rust/issues/98254

```

Comment:
This workaround is required for aarch rustc <= 1.72, but not for Rust 1.73 and later; and in fact will break on Rust 1.73 and later. I can either remove it (meaning it won't compile with Rust < 1.73), or I can introduce a dependency on "rustversion" to conditionally compile it based on compiler version. Thoughs?

Reply:
I like the condition for now. Makes it explicit and self-documented to deal with later.

Reply:
added the condition